python/tests/test_eclfile.py

```diff
@@ -0,0 +1,125 @@
+import unittest
+import sys
+import numpy as np
+
+from opm.io.ecl import EclFile
+
+
+class TestEclFile(unittest.TestCase):
+    
+    def test_getListOfArrays(self):
+
+        refList=["INTEHEAD","LOGIHEAD","DOUBHEAD","PORV","DEPTH","DX","DY","DZ","PORO",
+                 "PERMX","PERMY", "PERMZ","NTG","TRANX","TRANY","TRANZ","TABDIMS","TAB",
+                 "ACTNUM","EQLNUM","FIPNUM","PVTNUM","SATNUM","TRANNNC"]
+    
+        self.assertRaises(ValueError, EclFile, "/file/that/does_not_exists")
+
+        file2uf = EclFile("tests/data/SPE9.INIT")
+        self.assertEqual(file2uf.getNumArrays(), 24)
+
+        arrList = file2uf.getListOfArrays()
+        self.assertEqual(arrList, refList)
+
+        file2f = EclFile("tests/data/SPE9.FINIT")
+        self.assertEqual(file2f.getNumArrays(), 24)
+
+    def test_get_function(self):
+
+        file1 = EclFile("tests/data/SPE9.INIT")
+
+        first = file1.get(0)
+        self.assertEqual(len(first), 95)
+
+        # get fourth array in file SPE9.INIT which is PORV
+        test1 = file1.get(3)
+        test2 = file1.get("PORV")
+
+        for val1, val2 in zip(test1, test2):
+            self.assertEqual(val1, val2)
+
+    def test_get_function_float(self):
+
+        file1 = EclFile("tests/data/SPE9.INIT")
+         
+        dzList=[20.0, 15.0, 26.0, 15.0, 16.0, 14.0, 8.0, 8.0, 18.0, 12.0, 19.0, 18.0, 20.0, 50.0, 100.0]
+        poroList = [0.087, 0.097, 0.111, 0.16, 0.13, 0.17, 0.17, 0.08, 0.14, 0.13, 0.12, 0.105, 0.12, 0.116, 0.157]
+        ft3_to_bbl = 0.1781076
+        
+        refporv = []
+        
+        for poro, dz in zip(dzList, poroList):
+            for i in range(0,600):
+                refporv.append(300.0*300.0*dz*poro*ft3_to_bbl)
+        
+        self.assertTrue(file1.hasArray("PORV"))
+        porv_np = file1.get("PORV")
+        
+        self.assertEqual(len(porv_np), 9000)
+
+        self.assertTrue(isinstance(porv_np, np.ndarray))
+        self.assertEqual(porv_np.dtype, "float32")
+
+        porv_list = file1.get("PORV")
```

Comment:
I don't understand this API; can the `file.get()` method return *either* one vector *or* a list?